torchgeo/datasets/hyspecnet.py

```diff
@@ -0,0 +1,229 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+"""HySpecNet dataset."""
+
+import os
+from collections.abc import Callable, Sequence
+from typing import ClassVar
+
+import rasterio as rio
+import torch
+from einops import rearrange
+from matplotlib import pyplot as plt
+from matplotlib.figure import Figure
+from torch import Tensor
+
+from .errors import DatasetNotFoundError, RGBBandsMissingError
+from .geo import NonGeoDataset
+from .utils import Path, download_url, extract_archive, percentile_normalization
+
+# https://git.tu-berlin.de/rsim/hyspecnet-tools/-/blob/main/tif_to_npy.ipynb
+invalid_channels = [
+    126,
+    127,
+    128,
+    129,
+    130,
+    131,
+    132,
+    133,
+    134,
+    135,
+    136,
+    137,
+    138,
+    139,
+    140,
+    160,
+    161,
+    162,
+    163,
+    164,
+    165,
+    166,
+]
+valid_channels_ids = [c + 1 for c in range(224) if c not in invalid_channels]
+
+
+class HySpecNet11k(NonGeoDataset):
+    """HySpecNet-11k dataset.
+
+    `HySpecNet-11k <https://doi.org/10.5061/dryad.fttdz08zh>`__ is a large-scale
+    hyperspectral benchmark dataset made up of 11,483 nonoverlapping image patches
```

Comment:
From first inspection it's not clear what this dataset is a "benchmark" for. In the paper they look at compression I believe. Mabye we can state that it can be useful for pretraining (as done with the config task tests) or others. Or just include that it is made up of "unlabeled nonoverlapping image patches".

Reply:
Yeah, I also didn't know what to put in the CSV file for task. We don't currently have an SSL task in that table.